x/auth/tx/service.go

```diff
@@ -248,10 +248,58 @@ func (s txServer) GetBlockWithTxs(ctx context.Context, req *txtypes.GetBlockWith
 	}, nil
 }
 
+// BroadcastTx implements the ServiceServer.BroadcastTx RPC method.
 func (s txServer) BroadcastTx(ctx context.Context, req *txtypes.BroadcastTxRequest) (*txtypes.BroadcastTxResponse, error) {
 	return client.TxServiceBroadcast(ctx, s.clientCtx, req)
 }
 
+// TxEncode implements the ServiceServer.TxEncode RPC method.
+func (s txServer) TxEncode(ctx context.Context, req *txtypes.TxEncodeRequest) (*txtypes.TxEncodeResponse, error) {
+	if req.Tx == nil {
+		return nil, status.Error(codes.InvalidArgument, "invalid empty tx")
+	}
+
+	txb, err := s.clientCtx.Codec.Marshal(req.Tx)
+	if err != nil {
+		return nil, err
+	}
+
+	decodedBytes, err := s.clientCtx.TxConfig.TxDecoder()(txb)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
```suggestion
txBuilder := &wrapper{tx: req.Tx}
```

Why are we encoding, decoding, then encoding again?